### DISCUSSION TITLE
Why nerdctl can create duplicate containers

### DISCUSSION THREAD

**Discussion by llody55** [containerd/nerdctl#2584] [Q&A]
### Description

I created two containers with the same name when running two identical nerdctl run commands. Is this allowed?

### Steps to reproduce the issue

1.Create a container
`nerdctl run  -d --hostname rabbitmq-1-244  --name rabbitmq-1-244 -e RABBITMQ_DEFAULT_USER="admin" -e RABBITMQ_DEFAULT_PASS="admin" -e RABBITMQ_DEFAULT_VHOST=/ --restart=always -p "4369:4369"  -p "5672:5672"  -p "15672:15672"  -p "25672:25672" -p"15692:15692"   -v /data/rabbitmq/:/var/lib/rabbitmq  arm64v8/rabbitmq:3.9.11-management`
2.Create another identical container
`nerdctl run  -d --hostname rabbitmq-1-244  --name rabbitmq-1-244 -e RABBITMQ_DEFAULT_USER="admin" -e RABBITMQ_DEFAULT_PASS="admin" -e RABBITMQ_DEFAULT_VHOST=/ --restart=always -p "4369:4369"  -p "5672:5672"  -p "15672:15672"  -p "25672:25672" -p"15692:15692"   -v /data/rabbitmq/:/var/lib/rabbitmq  arm64v8/rabbitmq:3.9.11-management`

### Describe the results you received and expected

You should be prompted that the container name is duplicated or something like that.
![c26f247eaa4a95f08e213f297565849](https://github.com/containerd/nerdctl/assets/9710645/bf508434-ff0e-45f0-bae1-dcf3cf9b4297)

### What version of nerdctl are you using?

nerdctl version 1.5.0
containerd-1.6.10-linux-arm64

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

[root@node1 ~]# uname -a
Linux node1 5.10.0-153.27.0.103.oe2203sp2.aarch64 #1 SMP Wed Sep 13 19:04:10 CST 2023 aarch64 aarch64 aarch64 GNU/Linux
[root@node1 ~]# nerdctl -v
nerdctl version 1.5.0  
[root@node1 ~]# ctr -v
ctr github.com/containerd/containerd v1.6.10

**Comment by AkihiroSuda** [comment]
Explicitly allowed:
- https://github.com/containerd/nerdctl/blob/a1527794823dc6366ce64bb992bf29dfc5dd4366/pkg/labels/labels.go#L28-L30


The ID has to be unique, but the name does not.

**Reply by llody55** [reply]
In my experiment, I ran two identical commands. Not only did the container names have the same name, but the port mappings were also repeated. Is this allowed?

**Reply by fahedouch** [reply]
host exposed ports ?

**Reply by llody55** [reply]
@fahedouch Yes, the two commands are exactly the same and map the port to the host at the same time.
